ProjectLighthouse.Servers.GameServer/Controllers/MessageController.cs

```diff
@@ -55,16 +55,31 @@ public async Task<IActionResult> Announce()
     {
         GameTokenEntity token = this.GetToken();
 
-        string username = await this.database.UsernameFromGameToken(token);
+        UserEntity? user = await this.database.UserFromGameToken(token);
+        if (user == null) return this.BadRequest();
 
         StringBuilder announceText = new(ServerConfiguration.Instance.AnnounceText);
 
-        announceText.Replace("%user", username);
+        announceText.Replace("%user", user.Username);
         announceText.Replace("%id", token.UserId.ToString());
 
         if (ServerConfiguration.Instance.UserGeneratedContentLimits.ReadOnlyMode)
         {
-            announceText.Insert(0, BaseLayoutStrings.ReadOnlyWarn.Translate(LocalizationManager.DefaultLang) + "\n\n");
+            announceText.Append(BaseLayoutStrings.ReadOnlyWarn.Translate(LocalizationManager.DefaultLang) + "\n\n");
+        }
```

Comment:
Why was this changed? I thought the read-only warning was supposed to be a header